nemo/collections/asr/models/confidence_ensemble.py

```diff
@@ -12,22 +12,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from dataclasses import dataclass
 from typing import Dict, List, Optional, Union
 
 import joblib
 import numpy as np
 import torch
-from omegaconf import DictConfig, OmegaConf, open_dict
+from omegaconf import DictConfig, open_dict
 from pytorch_lightning import Trainer
 
 from nemo.collections.asr.models.asr_model import ASRModel
 from nemo.collections.asr.models.hybrid_rnnt_ctc_models import EncDecHybridRNNTCTCModel
-from nemo.collections.asr.parts.utils.asr_confidence_utils import ConfidenceConfig, get_confidence_aggregation_bank
+from nemo.collections.asr.parts.utils.asr_confidence_utils import (
+    ConfidenceConfig,
+    ConfidenceMethodConfig,
+    get_confidence_aggregation_bank,
+    get_confidence_measure_bank,
+)
 from nemo.collections.asr.parts.utils.audio_utils import ChannelSelectorType
 from nemo.core.classes import ModelPT
 from nemo.utils import model_utils
 
-__all__ = ['ConfidenceEnsembleModel']
+
+@dataclass(frozen=True)
```

Comment:
What's the frozen for ?

Reply:
It's required to be able to use it as a dictionary key in confidence tuning. Will add a comment